hrms/hr/doctype/expense_claim/bulk_download.py

```diff
@@ -0,0 +1,172 @@
+import frappe
+from frappe import _
+from frappe.utils.csvutils import build_csv_response
+import pandas as pd
+from suite42_erp_app.suite42_erp_app.common.decorators import (
+    handle_exceptions_with_readable_message,
+)
+import json
+from datetime import datetime
+
+
+@frappe.whitelist()
+@handle_exceptions_with_readable_message
+def bulk_download():
+    expense_claim_list = []
+    employee_list = frappe.db.get_list(
+        "Employee", filters={"status": "active"}, pluck="name", ignore_permissions=True
+    )
+    for employee_name in employee_list:
+        if employee_name != "HR-EMP-00015":
+            continue
+
+        pending_expense_claims = frappe.db.get_list(
+            "Expense Claim",
+            filters={
+                "employee": employee_name,
+                "status": "Pending Payment",
+                "grand_total": (">", 0),
+            },
+            fields=["name", "grand_total"],
+        )
+
+        # query = """
+        #     SELECT SUM(`grand_total`) AS total_amount
+        #     FROM `tabExpense Claim`
+        #     WHERE `employee` = %(employee_name)s
+        #     AND `status` = 'Pending Payment'
+        #     AND `grand_total` > 0
+        # """
+        # total_amount = frappe.db.sql(
+        #     query, {"employee_name": employee_name}, pluck="total_amount", as_dict=True
+        # )
+
+        if not pending_expense_claims:
+            continue
+
+        total_amount = 0
+        expense_claim_names = []
+        for row in pending_expense_claims:
+            expense_claim_names.append(row.get("name"))
+            total_amount += row.get("grand_total")
+
+        # if not total_amount:
+        #     continue
+        # total_amount = total_amount[0]
+
+        employee_doc = frappe.get_doc("Employee", employee_name)
+
+        # compay bank Account
+        company_bank_account_exists = frappe.db.get_list(
+            "Bank Account",
+            filters={"currency": "INR", "is_company_account": 1, "company": employee_doc.company},
+            pluck="name",
+        )
+        if not company_bank_account_exists:
+            frappe.throw(_(f"For Company {employee_doc.company} No INR Bank Account present"))
+        else:
+            company_bank_account_doc = frappe.get_doc(
+                "Bank Account", company_bank_account_exists[0]
+            )
+
+        # Employee Bank Acccount
+        employee_bank_account_exists = frappe.db.get_list(
+            "Bank Account",
+            filters={
+                "currency": "INR",
+                "company": employee_doc.company,
+                "party_type": "Employee",
+                "party": employee_name,
+            },
+            pluck="name",
+        )
+        if not employee_bank_account_exists:
+            frappe.throw(_(f"For Company {employee_doc.company} No INR Bank Account present"))
```

Comment:
Error msg is Employee account doesn't exist

Reply:
Done